arangod/RocksDBEngine/RocksDBValue.h

```diff
@@ -129,7 +129,7 @@ class RocksDBValue {
       : _type(other._type), _buffer(std::move(other._buffer)) {}
 
   RocksDBValue& operator=(RocksDBValue&& other) noexcept {
-    TRI_ASSERT(_type == other._type);
+    _type = other._type;
```

Comment:
Is this change still required? If not, I would like to revert it.

Reply:
Yes, it's needed to allow for holding a `RocksDBValue` buffer which is of unknown type at time of allocation. I'll add an assertion to ensure it is either assigning a value of the same type or that the initial type was `Placeholder`.